packages/core/core/src/Validation.js

```diff
@@ -0,0 +1,214 @@
+      return loadConfig(configRequest, parentNodeId);
+    };
+    this.options = options;
+  }
+
+  async run(): Promise<void> {
+    if (this.request.filePath.includes('node_modules')) return;
+
+    report({
+      type: 'validation',
+      request: this.request
+    });
+
+    this.cache = new Cache(this.options.outputFS, this.options.cacheDir);
+
+    let asset = await this.loadAsset();
+    let pipeline = await this.loadPipeline(asset.filePath);
+    await pipeline.run(asset);
+  }
+
+  async loadAsset(): Promise<InternalAsset> {
+    let {filePath, env, code, sideEffects} = this.request;
+    let {content, size, hash} = await summarizeRequest(
+      this.options.inputFS,
+      this.request
+    );
+
+    return new InternalAsset({
+      // If the transformer request passed code rather than a filename,
+      // use a hash as the base for the id to ensure it is unique.
+      idBase: code != null ? hash : filePath,
+      fs: this.options.inputFS,
+      filePath: filePath,
+      type: path.extname(filePath).slice(1),
+      cache: this.cache,
+      ast: null,
+      content,
+      hash,
+      env: env,
+      stats: {
+        time: 0,
+        size
+      },
+      sideEffects: sideEffects
+    });
+  }
+
+  async loadPipeline(filePath: FilePath): Promise<Pipeline> {
+    let configRequest = {
+      filePath,
+      meta: {
+        actionType: 'validation'
+      }
+    };
+    let configs = new Map();
+
+    let config = await this.loadConfig(configRequest);
+    let parcelConfig = nullthrows(config.result);
+
+    configs.set('parcel', config);
+
+    let pipeline = new Pipeline({
+      validators: await parcelConfig.getValidators(filePath),
+      configs,
+      options: this.options
+    });
+
+    return pipeline;
+  }
+}
+
+type PipelineOpts = {|
+  validators: Array<Validator>,
+  configs: ConfigMap,
+  options: ParcelOptions
+|};
+
+class Pipeline {
+  validators: Array<Validator>;
+  configs: ConfigMap;
+  options: ParcelOptions;
+  resolverRunner: ResolverRunner;
+  generate: GenerateFunc;
+  postProcess: ?PostProcessFunc;
+
+  constructor({validators, configs, options}: PipelineOpts) {
+    this.validators = validators;
+    this.configs = configs;
+    this.options = options;
+    let parcelConfig = nullthrows(this.configs.get('parcel'));
+    parcelConfig = nullthrows(parcelConfig.result);
+    this.resolverRunner = new ResolverRunner({
+      config: parcelConfig,
+      options
+    });
+  }
+
+  async run(asset: InternalAsset): Promise<void> {
+    const resolve = async (from: FilePath, to: string): Promise<FilePath> => {
+      return (await this.resolverRunner.resolve(
+        new Dependency({
+          env: asset.env,
+          moduleSpecifier: to,
+          sourcePath: from
+        })
+      )).filePath;
+    };
+
+    for (let validator of this.validators) {
+      await validator.validate({
+        asset: new MutableAsset(asset),
+        options: this.options,
+        resolve
```

Comment:
Not sure, Would be nice to overwrite the real TS resolver with whatever resolver Parcel uses. Although not sure if it's a good idea because that might give unexpected behaviour compared to running tsc

Reply:
Good question...

Reply:
Removed the resolve function for now